File: invesdwin-context-parent/invesdwin-context-clojure/src/test/java/de/invesdwin/context/clojure/tests/InputsAndResultsTestLong.clj
(defmacro defined [& var]
    `(try 
        (or 
            (boolean (resolve ~@var)) 
            (not (nil? (eval ~@var)))
        ) (catch Exception e# false)
    )
)

(println "getLong")
(if (defined 'getLong)
	(throw (Exception. "getLong already defined!"))
)
(def getLong putLong)
(def getLongType (class getLong))
(println getLongType)
(println getLong)
(if-not (= getLongType Long)
	(throw (Exception. "getLong not Long!"))
)

(println "getLongVector")
(if (defined 'getLongVector)
	(throw (Exception. "getLongVector already defined!"))
)
(def getLongVector putLongVector)
(def getLongVectorType (class (get getLongVector 0)))
(println getLongVectorType)
(println getLongVector)
(if-not (= getLongVectorType Long)
	(throw (Exception. "getLongVector not Long!"))
)

(println "getLongVectorAsList")
(if (defined 'getLongVectorAsList)
	(throw (Exception. "getLongVectorAsList already defined!"))
)
(def getLongVectorAsList putLongVectorAsList)
(def getLongVectorAsListType (class (get getLongVectorAsList 0)))
(println getLongVectorAsListType)
(println getLongVectorAsList)
(if-not (= getLongVectorAsListType Long)
	(throw (Exception. "getLongVectorAsList not Long!"))
)

(println "getLongMatrix")
(if (defined 'getLongMatrix)
	(throw (Exception. "getLongMatrix already defined!"))
)
(def getLongMatrix putLongMatrix)
(def getLongMatrixType (class (get (get getLongMatrix 0) 0)))
(println getLongMatrixType)
(println getLongMatrix)
(if-not (= getLongMatrixType Long)
	(throw (Exception. "getLongMatrix not Long!"))
)

(println "getLongMatrixAsList")
(if (defined 'getLongMatrixAsList)
	(throw (Exception. "getLongMatrixAsList already defined!"))
)
(def getLongMatrixAsList putLongMatrixAsList)
(def getLongMatrixAsListType (class (get (get getLongMatrixAsList 0) 0)))
(println getLongMatrixAsListType)
(println getLongMatrixAsList)
(if-not (= getLongMatrixAsListType Long)
	(throw (Exception. "getLongMatrixAsList not Long!"))
)

File: invesdwin-context-parent/invesdwin-context-ruby-contract/src/test/java/de/invesdwin/context/ruby/tests/InputsAndResultsTestLong.rb
puts("getLong")
unless (defined?(getLong)).nil? || getLong.nil?
	raise "getLong already defined!"
end
getLong = putLong
getLongType = getLong.class
puts(getLongType)
puts(getLong)
if(getLongType != Integer)
	raise "getLong not Integer!"
end

puts("getLongVector")
unless (defined?(getLongVector)).nil? || getLongVector.nil?
	raise "getLongVector already defined!"
end
getLongVector = putLongVector
getLongVectorType = getLongVector[0].class
puts(getLongVectorType)
puts(getLongVector)
if(getLongVectorType != Integer)
	raise "getLongVector not Integer!"
end

puts("getLongVectorAsList")
unless (defined?(getLongVectorAsList)).nil? || getLongVectorAsList.nil?
	raise "getLongVectorAsList already defined!"
end
getLongVectorAsList = putLongVectorAsList
getLongVectorAsListType = getLongVectorAsList[0].class
puts(getLongVectorAsListType)
puts(getLongVectorAsList)
if(getLongVectorAsListType != Integer)
	raise "getLongVectorAsList not Integer!"
end

puts("getLongMatrix")
unless (defined?(getLongMatrix)).nil? || getLongMatrix.nil?
	raise "getLongMatrix already defined!"
end
getLongMatrix = putLongMatrix
getLongMatrixType = getLongMatrix[0][0].class
puts(getLongMatrixType)
puts(getLongMatrix)
if(getLongMatrixType != Integer)
	raise "getLongMatrix not Integer!"
end

puts("getLongMatrixAsList")
unless (defined?(getLongMatrixAsList)).nil? || getLongMatrixAsList.nil?
	raise "getLongMatrixAsList already defined!"
end
getLongMatrixAsList = putLongMatrixAsList
getLongMatrixAsListType = getLongMatrixAsList[0][0].class
puts(getLongMatrixAsListType)
puts(getLongMatrixAsList)
if(getLongMatrixAsListType != Integer)
	raise "getLongMatrixAsList not Integer!"
end
